Sources/ETCoordinator/Starter.swift

```diff
@@ -68,24 +72,35 @@ open class Starter: NSObject {
 // MARK: - Navigation Controller Delegate + Injection
 
 extension Starter: UIAdaptivePresentationControllerDelegate {
+    public func presentationControllerWillDismiss(_ presentationController: UIPresentationController) {
+        isDismissing = true
+    }
+
+    public func presentationControllerDidAttemptToDismiss(_ presentationController: UIPresentationController) {
+        isDismissing = false
+    }
+
+
     public func presentationControllerDidDismiss(_ presentationController: UIPresentationController) {
         guard isStarted else {
             return
         }
 
         didStopWithGestureVC()
+        isDismissing = false
     }
 }
 
 extension Starter: UINavigationControllerDelegate {
     @objc public func navigationController(_ nc: UINavigationController, didShow vc: UIViewController, animated: Bool) {
-        guard isStarted else {
+        guard isStarted && !isDismissing else {
             return
         }
         if let didStartCompletion = didStartCompletion, didStartCompletion.vc === vc {
             didStartCompletion.action()
             self.didStartCompletion = nil
         }
+        // This is called when stoping (poping) pushed coordinator.
```

Comment:
@urbanp11 
This if caused the memory leak while pushing coordinators. I removed the `didStopWithGestureVC` method call in some of previous commits so now I'm putting it back.